radix-engine/src/blueprints/account/blueprint.rs

```diff
@@ -89,32 +89,36 @@ impl AccountBlueprint {
         Ok(modules)
     }
 
-    pub fn create_virtual_secp256k1<Y>(
-        input: VirtualLazyLoadInput,
+    pub fn on_virtualize<Y>(
+        input: OnVirtualizeInput,
         api: &mut Y,
-    ) -> Result<VirtualLazyLoadOutput, RuntimeError>
+    ) -> Result<OnVirtualizeOutput, RuntimeError>
     where
         Y: ClientApi<RuntimeError>,
     {
-        let public_key_hash = PublicKeyHash::Secp256k1(Secp256k1PublicKeyHash(input.id));
-        Self::create_virtual(public_key_hash, api)
-    }
-
-    pub fn create_virtual_ed25519<Y>(
-        input: VirtualLazyLoadInput,
-        api: &mut Y,
-    ) -> Result<VirtualLazyLoadOutput, RuntimeError>
-    where
-        Y: ClientApi<RuntimeError>,
-    {
-        let public_key_hash = PublicKeyHash::Ed25519(Ed25519PublicKeyHash(input.id));
-        Self::create_virtual(public_key_hash, api)
+        match input.node_id.entity_type() {
```

Comment:
I much prefer the application layer being unaware of entity types (should be a system layer invariant). Is there a reason why this was moved into this layer?

Reply:
This is because there is only one blueprint (which implies one `on_virtualize` hook) for account.

I don't imagine we'd like `Ed25519Account` and `Secp256k1Account` lol.